package/MDAnalysis/analysis/base.py

```diff
@@ -125,14 +125,192 @@
 import inspect
 import logging
 import itertools
+import warnings
 
 import numpy as np
 from MDAnalysis import coordinates
 from MDAnalysis.core.groups import AtomGroup
 from MDAnalysis.lib.log import ProgressBar
+from MDAnalysis.lib.util import is_installed
+from typing import Callable, Iterable
+from importlib import import_module
+from functools import partial
 
 logger = logging.getLogger(__name__)
 
+def is_valid_distributed_client(client: object):
+    try:
+        import dask.distributed as distributed
+    except ImportError:
+        return False
+    return isinstance(client, distributed.Client)
+
+class ParallelExecutor:
```

Comment:
```suggestion

class ParallelExecutor:
```

A 1k+ line file is a lot. Is there a point at which we should consider moving this class out of here and into it's own file?

Reply:
I think it makes sense to create a new file e.g. `parallel.py` inside analysis and move this class there.

Reply:
Yes, I agree, especially if it becomes a module that doesn't need to import anything else from MDA.

Reply:
The new file is there now

Reply:
Solving because things have been split out to `parallel.py`. @marinegor please resolve conversations where the it is trivial to agree that it's solved.